### ISSUE TITLE
Combine DragHandleAutoScroller into AutoScroller

### ISSUE THREAD

**Issue by matthew-carroll** [superlistapp/super_editor#546]
We have a generic auto-scrolling object called `AutoScroller`. It contains methods for animating scrolling over time.

We also have `DragHandleAutoScroller`, which, despite the name, doesn't have an API that's specific to drag handles. 

I think that `DragHandleAutoScroller` should be merged into `AutoScroller` as a generic auto-scrolling solution.

Also, add appropriate tests for the auto-scroller.

**Comment by ThangVuNguyenViet** [comment]
Hi @matthew-carroll . I took a look and find that "assigning and unassigning" the `scrollPosition` in the `AutoScroller` is a bit redundant. I briefly "merge" the two classes and it seems to work with getting the `scrollPosition` directly from the method. Is there anything special with the "assigning and unassigning" thing? Or else we should remove it.

BTW, should a PR for this issue depends on #533 ?

**Comment by matthew-carroll** [comment]
Please don't work on this yet. I haven't decided for sure what should happen here. I'll tag you on any issue that I'd like to assign, or you can reach out about a ticket that you think you should work on.
